pkg/operator/k8sutil/configmap.go

```diff
@@ -62,13 +64,31 @@ func GetOperatorSetting(context context.Context, clientset kubernetes.Interface,
 }
 
 func GetValue(data map[string]string, settingName, defaultValue string) string {
-	if settingValue, ok := data[settingName]; ok {
-		logger.Infof("%s=%q (configmap)", settingName, settingValue)
-		return settingValue
-	} else if settingValue, ok := os.LookupEnv(settingName); ok {
-		logger.Infof("%s=%q (env var)", settingName, settingValue)
+	var settingValue, settingSource string
+
+	if val, ok := data[settingName]; ok {
+		settingValue = val
+		settingSource = "configmap"
+	} else if val, ok := os.LookupEnv(settingName); ok {
+		settingValue = val
+		settingSource = "env var"
+	}
+
+	if settingValue != "" {
+		if logChangedSettings(settingName, settingValue, settingSource) {
+			logger.Infof("%s=%q (%s)", settingName, settingValue, settingSource)
+		}
 		return settingValue
 	}
+
 	logger.Infof("%s=%q (default)", settingName, defaultValue)
```

Comment:
Can we also make it so the default is only logged once? Maybe this default setting just needs to be an `else` block up on line 75.

Reply:
yes, updated